server-spi-private/src/main/java/org/keycloak/cookie/CookieType.java

```diff
@@ -55,4 +93,42 @@ public Integer getDefaultMaxAge() {
         return defaultMaxAge;
     }
 
+    private static class CookieTypeBuilder {
+
+        private String name;
+        private boolean supportSameSiteLegacy = false;
```

Comment:
Should we not be defaulting this to `true` for now?

Reply:
Nah, because it's an opt-in for specific cookie types.